applications/asset_tracker/src/cloud_codec/service_info.c

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+#include "service_info.h"
+
+#define SERVICE_INFO_JSON_NAME "serviceInfo"
+#define SENSORS_JSON_NAME "sensors"
+#define FOTAS_JSON_NAME "fota_v1"
+
+static int add_array_obj(const char * const items[], const u32_t item_cnt,
+			 const char * const item_name, cJSON * const obj)
+{
+	if (!obj || !item_name) {
+		return -EINVAL;
+	}
+
+	cJSON * array = cJSON_CreateArray();
+
+	if (!array) {
+		return -ENOMEM;
+	}
+
+	for (u32_t cnt = 0; items && (cnt < item_cnt); ++cnt) {
+		if (items[cnt]) {
+			cJSON_AddItemToArray(array,
+					     cJSON_CreateString(items[cnt]));
+		}
+	}
+
+	cJSON_AddItemToObject(obj, item_name, array);
+
+	return 0;
+}
+
+int service_info_json_object_encode(const char * const sensors[],
+				    const u32_t sensor_count,
+				    const char * const fota[],
+				    const u32_t fota_count,
+				    cJSON * const obj_out)
+{
+	if (obj_out == NULL || (!sensors && sensor_count) ||
```

Comment:
Please unify how you check for NULL pointers. Personally I prefer an explicit comparison, but `!ptr` is fine as well, just be consistent.